core/src/main/java/hudson/security/ACLContext.java

```diff
@@ -52,12 +52,21 @@
     /**
      * Accessor for the previous context.
      * @return the previous context.
+     * @since TODO
      */
     @NonNull
-    public SecurityContext getPreviousContext() {
+    public SecurityContext getPreviousContext2() {
```

Comment:
does this even need to be exposed (could be said that it is just a poilluting methid in the previous version) and the only legitimate use case is in `close`

Reply:
use case in https://github.com/jenkinsci/jenkins/pull/4848/files#diff-7703322191e82930e0d5d450dc9ba1adR322

Reply:
Probably true, AFAIK it is unused.

Reply:
Ah, not quite, used inside core at least: https://github.com/jenkinsci/jenkins/blob/7edfc7170c56780ccc715ed67cbe58a37d65bafb/core/src/main/java/hudson/model/AbstractItem.java#L316-L327

Reply:
just that one occurrence in Jenkins.  basically backtracking authentication after elevation when an operation failed (was checking if an item existed, which needs to run as the system user).

Reply:
Right. Anyway, I can mark this restricted, or just leave it as is—harmless, reasonably intuitive getter.